File: app/src/main/java/com/wangbin/mydome/activity/MainActivity.java
package com.wangbin.mydome.activity;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.design.widget.NavigationView;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.MenuItem;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;

import com.wangbin.mydome.R;
import com.wangbin.mydome.adapter.SearchAdapter;
import com.wangbin.mydome.adapter.ViewPagerAdapter;
import com.wangbin.mydome.bean.SearchBean;
import com.wangbin.mydome.fragment.NewsFragment;
import com.wangbin.mydome.fragment.TestFragment;
import com.wangbin.mydome.helper.BottomNavigationViewHelper;

import java.util.ArrayList;
import java.util.List;


/**
 * 主界面
 */
public class MainActivity extends AppCompatActivity {
    private BottomNavigationView bottomNavigationView;
    private NavigationView nav_view;
    private DrawerLayout drawer_layout;
    private RecyclerView rec_search;
    private EditText et_search;
    private ImageView img_search;
    private ViewPagerAdapter viewPagerAdapter;
    private SearchAdapter searchAdapter;
    private ViewPager viewPager;
    private MenuItem menuItem;
    private int num = 1;
    private List<SearchBean> datas = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
        initLinster();
    }

    private void initView() {
        bottomNavigationView = findViewById(R.id.bottomNavigationView);
        bottomNavigationView.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
        BottomNavigationViewHelper.disableShiftMode(bottomNavigationView);
        viewPager = findViewById(R.id.vp);
        viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                if (menuItem != null) {
                    menuItem.setChecked(false);
                } else {
                    bottomNavigationView.getMenu().getItem(0).setChecked(false);
                }
                menuItem = bottomNavigationView.getMenu().getItem(position);
                menuItem.setChecked(true);
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });
        viewPagerAdapter = new ViewPagerAdapter(getSupportFragmentManager());
        viewPager.setAdapter(viewPagerAdapter);
        List<Fragment> list = new ArrayList<>();
        list.add(NewsFragment.newInstance("首页"));
        list.add(NewsFragment.newInstance("公告"));
        list.add(TestFragment.newInstance("趣闻"));
        list.add(TestFragment.newInstance("个人"));
        viewPagerAdapter.setList(list);

        drawer_layout = findViewById(R.id.drawer_layout);
        nav_view = findViewById(R.id.nav_view);
        View headerView = nav_view.getHeaderView(0);
        rec_search = headerView.findViewById(R.id.rec_search);
        et_search = headerView.findViewById(R.id.et_search);
        img_search = headerView.findViewById(R.id.img_search);

        LinearLayoutManager layoutManager = new LinearLayoutManager(MainActivity.this);
        rec_search.setLayoutManager(layoutManager);
        searchAdapter = new SearchAdapter(MainActivity.this, datas);
        rec_search.setAdapter(searchAdapter);
    }

    private void initLinster() {
        img_search.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                datas.clear();
                for (int i = 0; i < 5; i++) {
                    SearchBean searchBean = new SearchBean();
                    searchBean.setName("测试数据" + num);
                    searchBean.setFlag(false);
                    datas.add(searchBean);
                    num++;
                }
                searchAdapter.notifyDataSetChanged();
            }
        });

        searchAdapter.setOnItemClickLitener(new SearchAdapter.OnItemClickListener() {
            @Override
            public void onItemClick(View view, int position) {
                for (SearchBean searchBean : datas) {
                    searchBean.setFlag(false);
                }
                datas.get(position).setFlag(true);
                searchAdapter.notifyDataSetChanged();
            }
        });

    }

    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener = new BottomNavigationView.OnNavigationItemSelectedListener() {

        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            menuItem = item;
            switch (item.getItemId()) {
                case R.id.navigation_home:
                    viewPager.setCurrentItem(0);
                    return true;
                case R.id.navigation_dashboard:
                    viewPager.setCurrentItem(1);
                    return true;
                case R.id.navigation_notifications:
                    viewPager.setCurrentItem(2);
                    return true;
                case R.id.navigation_person:
                    viewPager.setCurrentItem(3);
                    return true;
            }
            return false;
        }
    };

}

File: app/src/main/java/com/wangbin/mydome/activity/MainActivity.kt
package com.wangbin.mydome.activity

import android.support.design.widget.BottomNavigationView
import android.support.v4.app.Fragment
import android.support.v4.view.ViewPager
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.MenuItem
import android.view.View
import android.widget.ImageView
import com.vector.update_app.UpdateAppManager
import com.wangbin.mydome.adapter.SearchAdapter
import com.wangbin.mydome.adapter.ViewPagerAdapter
import com.wangbin.mydome.base.BaseActivity
import com.wangbin.mydome.bean.SearchBean
import com.wangbin.mydome.fragment.IndexFragment
import com.wangbin.mydome.fragment.TestFragment
import com.wangbin.mydome.helper.BottomNavigationViewHelper
import com.wangbin.mydome.net.UpdateAppHttpUtil
import com.wangbin.mydome.net.Url.Companion.url
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*


/**
 * 主界面
 */
class MainActivity : BaseActivity() {

    private var viewPagerAdapter: ViewPagerAdapter? = null
    private var searchAdapter: SearchAdapter? = null
    private var menuItem: MenuItem? = null
    private var num = 1
    private val mDatas = ArrayList<SearchBean>()
    private var titles: Array<String>? = null
    private var imgSearch: ImageView? = null

    override fun intiLayout(): Int {
        return com.wangbin.mydome.R.layout.activity_main
    }

    override fun widgetClick(view: View) {
    }

    private val mOnNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        menuItem = item
        when (item.itemId) {
            com.wangbin.mydome.R.id.navigation_home -> {
                viewPager!!.currentItem = 0
                return@OnNavigationItemSelectedListener true
            }
            com.wangbin.mydome.R.id.navigation_dashboard -> {
                viewPager!!.currentItem = 1
                return@OnNavigationItemSelectedListener true
            }
            com.wangbin.mydome.R.id.navigation_notifications -> {
                viewPager!!.currentItem = 2
                return@OnNavigationItemSelectedListener true
            }
            com.wangbin.mydome.R.id.navigation_person -> {
                viewPager!!.currentItem = 3
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    override fun initView() {
        titles = arrayOf(resources.getString(com.wangbin.mydome.R.string.title1), resources.getString(com.wangbin.mydome.R.string.title2),
                resources.getString(com.wangbin.mydome.R.string.title3), resources.getString(com.wangbin.mydome.R.string.title3))
        bottomNavigationView!!.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)
        BottomNavigationViewHelper.disableShiftMode(bottomNavigationView!!)
        viewPager!!.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {

            }

            override fun onPageSelected(position: Int) {
                if (menuItem != null) {
                    menuItem!!.isChecked = false
                } else {
                    bottomNavigationView!!.menu.getItem(0).isChecked = false
                }
                menuItem = bottomNavigationView!!.menu.getItem(position)
                menuItem!!.isChecked = true
            }

            override fun onPageScrollStateChanged(state: Int) {

            }
        })
        viewPagerAdapter = ViewPagerAdapter(supportFragmentManager)
        viewPager!!.adapter = viewPagerAdapter
        val list = ArrayList<Fragment>()
        list.add(IndexFragment.newInstance(titles!![0]))
        list.add(TestFragment.newInstance(titles!![1]))
        list.add(TestFragment.newInstance(titles!![2]))
        list.add(TestFragment.newInstance(titles!![3]))
        viewPagerAdapter!!.setList(list)

        val headerView = nav_view!!.getHeaderView(0)
        val recSearch = headerView.findViewById<RecyclerView>(com.wangbin.mydome.R.id.rec_search)
        imgSearch = headerView.findViewById(com.wangbin.mydome.R.id.img_search)

        recSearch!!.layoutManager = LinearLayoutManager(this@MainActivity)
        searchAdapter = SearchAdapter(this@MainActivity, mDatas)
        recSearch.adapter = searchAdapter
        UpdateAppManager.Builder()
                //当前 Activity
                .setActivity(this)
                //更新地址
                .setUpdateUrl(url.BASE_URL + url.UPDATE)
                //实现 httpManager 接口的对象
                .setHttpManager(UpdateAppHttpUtil())
                .build()
                .update()
    }

    override fun setListener() {
        imgSearch!!.setOnClickListener {
            mDatas.clear()
            for (i in 0..4) {
                val searchBean = SearchBean()
                searchBean.name = "测试数据$num"
                searchBean.isFlag = false
                mDatas.add(searchBean)
                num++
            }
            searchAdapter!!.notifyDataSetChanged()
        }

        searchAdapter!!.setOnItemClickLitener(object : SearchAdapter.OnItemClickListener {
            override fun onItemClick(view: View, position: Int) {
                for (searchBean in mDatas) {
                    searchBean.isFlag = false
                }
                mDatas[position].isFlag = true
                searchAdapter!!.notifyDataSetChanged()
            }
        })
    }

}
